### PR TITLE
Resource merge precedence

### LINKED ISSUE #1712: Reverse priority for Resource.Merge
Spec reversed the order of priority. The incoming resource take precedence in case of collision.
https://github.com/open-telemetry/opentelemetry-specification/pull/1345/files

### PR DESCRIPTION
**Pull request by Austin-Tan**
Fixes #1712 

## Changes

When merging an "other" resource to "this" resource, we previously kept the values from "this" resource in conflicts. Due to recent changes in the OTel specification, this has been reversed to always use the incoming resource's attributes.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,10 @@
   them to supported data types (long for int/short, double for float). For
   invalid attributes we now throw an exception instead of logging an error.
   ([#1720](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1720))
+* Merging "this" resource with another previously kept "this" resource's
```
Comment (CodeBlanch) [inline src/OpenTelemetry/CHANGELOG.md:29]: This sentence is crashing my brain 🤣 Did you mean to use "this" twice?

File: src/OpenTelemetry/Resources/Resource.cs [hunk 2]
```diff
@@ -57,34 +57,34 @@ internal Resource(IEnumerable<KeyValuePair<string, object>> attributes)
         public IEnumerable<KeyValuePair<string, object>> Attributes { get; }
 
         /// <summary>
-        /// Returns a new, merged <see cref="Resource"/> by merging the current <see cref="Resource"/> with the.
-        /// <code>other</code> <see cref="Resource"/>. In case of a collision the current <see cref="Resource"/> takes precedence.
+        /// Returns a new, merged <see cref="Resource"/> by merging the old <see cref="Resource"/> with the.
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Resources/Resource.cs:60]: nit: Looks like there's a period at the end of the first line that should be removed.

**Review by CodeBlanch — APPROVED** [review]
Left a couple of nits, but LGTM.

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (Austin-Tan) [reply src/OpenTelemetry/CHANGELOG.md:29]: Haha I did, but you're right that it is confusing and on a re-read it took me many tries to comprehend, I'll reword it.

File: src/OpenTelemetry/Resources/Resource.cs (hunk 2 above)
Comment (Austin-Tan) [reply src/OpenTelemetry/Resources/Resource.cs:60]: good catch, seems error SA1629 prompts you to add a period to the end of documentation text.

Comment (CodeBlanch) [reply src/OpenTelemetry/Resources/Resource.cs]: I think the issue is the `<code>` which creates a line break. Try switching it to `<c>other</c>` which will render inline.

**Pull request merged**